src/mds/Locker.cc

```diff
@@ -5414,3 +5411,17 @@ void Locker::handle_file_lock(ScatterLock *lock, const cref_t<MLock> &m)
     ceph_abort();
   }  
 }
+
+void Locker::lock_stablized(SimpleLock* lock, uint64_t mask, int r, MDSContext::vec* finishers) {
+  if (lock->get_type() == CEPH_LOCK_INEST && lock->is_dirty()) {
+    dout(20) << __func__ << " lock " << *lock << " on " << *lock->get_parent() 
+    << " pushed to the back of updated_scatterlocks due to quickflush" << dendl;
+    mark_updated_scatterlock(static_cast<ScatterLock*>(lock));
```

Comment:
why not do this in Locker::handle_file_lock() for LOCK_ACK_FOO cases

Reply:
@ukernel Hi, when the nestlock turns stable, in addition to making sure it is in updated_scatterlocks queue if it's dirty, we also have to clear the "NUDGING" state. So, instead of adding "clear_nudging" all over the place, I added a "lock_stablized" method and do all necessary works there.

Reply:
@ukernel Hi, thanks for your review:-)

Besides the handle_file_lock case that leave dirty nestlock outside updated_scatterlocks, there are also other cases, such as the following:

1. say an inode A's nestlock is being nudged, and an subdirectory B wants to propagate rstat up to A, it will mark A's nestlock dirty and put it in updated_scatterlocks;
2. before A's previous nudge completes, scatter_tick or other rstat propagation operation remove A's nestlock from updated_scatterlocks and tried to nudge it, which would fail because A's nestlock is not stable;
3. When A's previous nudge operation completes, its nestlock wouldn't be in the updated_scatterlocks queue, which means the subdirectory B's rstat propagation operation is lost.

So, to prevent all these cases from happening, I added this method to do the requeue whenever the nestlock turns stable.